Modules/mathmodule.c

```diff
@@ -3885,7 +3892,91 @@ math_nextafter_impl(PyObject *module, double x, double y)
         return PyFloat_FromDouble(y);
     }
 #endif
-    return PyFloat_FromDouble(nextafter(x, y));
+    if (steps == Py_None) {
+        // fast path: we default to one step.
+        return PyFloat_FromDouble(nextafter(x, y));
+    }
+    steps = PyNumber_Index(steps);
+    if (steps == NULL) {
+        return NULL;
+    }
+    assert(PyLong_CheckExact(steps));
+    if (_PyLong_IsNegative((PyLongObject *)steps)) {
+        PyErr_SetString(PyExc_ValueError,
+                        "steps must be a non-negative integer");
+        Py_DECREF(steps);
+        return NULL;
+    }
+
+    uint64_t usteps = PyLong_AsUnsignedLongLong(steps);
```

Comment:
This isn't quite portable: if `unsigned long long` has more than 64 bits, we lose information when converting to `uint64_t`, and then end up with the wrong answer (e.g., if `usteps = 2**64 + 1`).

I don't know of any current platform where `unsigned long long` has more than 64 bits, and it seems a reasonable bet that such a platform is unlikely in the future. As a quick fix, I'd suggest just adding a preprocessor check on `SIZEOF_LONG_LONG` so that if this code ever happens to meet such a platform, it'll fail noisily.

Reply:
Done in 66ef3f968037c693fec059aa85f4740bf4ddf04f